src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.ts

```diff
@@ -13,26 +21,29 @@ import {VariableService} from '../../../../services/variable.service';
 })
 export class GbVariablesComponent implements OnInit {
 
-  public allChecked: boolean;
+  private _allChecked: boolean;
+  public viewMode: VariablesViewMode;
+  public availableViewModes: SelectItem[];
 
   public VariablesViewMode = VariablesViewMode; // make enum visible in template
   public readonly fileElementId: string = 'variablesCriteriaFileUpload';
 
   public isUploadListenerNotAdded: boolean;
   public file: File; // holds the uploaded cohort file
 
-  private _availableViewModes: SelectItem[];
-
   constructor(private variableService: VariableService,
               private navbarService: NavbarService) {
-    this.allChecked = true;
     this.isUploadListenerNotAdded = true;
-    this.viewMode = VariablesViewMode.TREE_VIEW;
-    this.availableViewModes = this.listAvailableViewModes();
   }
 
   ngOnInit() {
-    this.checkAllVariables(true);
+    this.availableViewModes = Object.keys(VariablesViewMode).map(c => {
```

Comment:
Having logic to collect all possible select items in the separate function `listAvailableViewModes()` was not that bad idea.

Reply:
if `listAvailableViewModes()` is used just once and this function only contains a mapper function inside, is it still a good idea ? :)

Reply:
I think so.
it's +5 lines in the init block.
IMHO It takes slightly less time to read this code:
```
this.availableViewModes = listAvailableViewModes()
this.viewMode = VariablesViewMode.TREE_VIEW;
```
If I'd want to know how we fetch those modes I could always jump to the function definition.

But indeed it could be a matter of taste.